src/GenSequence.ts

```diff
@@ -282,6 +286,33 @@ export function first<T>(fn: (t: T) => boolean, defaultValue: T, i: Iterable<T>)
     return defaultValue;
 }
 
+export function max<T>(i: Iterable<T>): Maybe<T> {
+    let sawValue: boolean = false;
+    const value: Maybe<T> = reduce((p: T, c: T) => {
+        if (!sawValue) {
+            sawValue = true;
+        }
+        
+        if (isNonValue(p) || c > p) {
+            return c;
+        }
+
+        return p;
+    },
+    undefined, 
+    i);
+
+    if (!sawValue) {
+        throw new Error("the sequence is empty.");
+    }
+    
+    if (isNonValue(value)) {
+        return undefined;
+    }
+
+    return value;
+}
+
```

Comment:
Good point.

Your proposal is easy enough to read:
```typescript
const seq = genSequence([{name="joe", age=15}, {name="nancy", age=43}]);
const m = seq
    .max(v => v.age)
```

Yet, how would you compare a vector:

```typescript
const seq = genSequence([{x=45, y=2}, {x=45, y=10}]);
const m = seq
    .max((p, c) => c.x > p.x || (c.x == p.x && c.y > p.y))
```

Of course, if someone wants to do something that complicated, they can write their own reducer.

My vote is to go with your suggestion.  It is less powerful, but easier to read and get correct.  I think users of the function will find it more useful in the general case.